process/smartContract/process.go

```diff
@@ -1315,12 +1315,25 @@ func createBaseSCR(
 	return result
 }
 
+func addVMOutputResultsToSCR(vmOutput *vmcommon.VMOutput, result *smartContractResult.SmartContractResult) {
+	result.CallType = vmcommon.AsynchronousCallBack
+	result.GasLimit = vmOutput.GasRemaining
+	result.Data = []byte("@" + core.ConvertToEvenHex(int(vmOutput.ReturnCode)))
+
+	for _, retData := range vmOutput.ReturnData {
+		result.Data = append(result.Data, []byte("@"+hex.EncodeToString(retData))...)
+	}
```

Comment:
This could be made into a function. Some similar code is also found at the end of `createSCRForSenderAndRelayer()`.

Reply:
done